### ISSUE TITLE
Roslyn (Find all References and CodeLens References Indicator) incorrectly reports references to explicitly implemented interface members from other, non-explicit implementations

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#43460]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/800744/roslyn-find-all-references-and-codelens-references.html)._

---
Serious find-all-references "corner case" issue--

Expected: When Find-all-references and CodeLens References indicator are triggered for an explicitly implemented interface method (in a class), they report ONLY every other call to the method where it's explicitly implemented. References to non-explicit implementations of the interface method are NOT included.

Actual: When Find-all-references and CodeLens References indicator are triggered for an explicitly implemented interface method (in a class), they report every call to the interface method as a reference to that explicitly-implemented interface method. Regardless if each reference is actually to an explicit implementation of the queried interface method. These results are identical to results of running find-all-references on the interface definition itself.

---
#### Visual Studio Feedback System on 10/30/2019, 07:26 PM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

#### Visual Studio Feedback System on 10/31/2019, 04:30 PM: 

<p>This issue is currently being investigated. Our team will get back to you if either more information is needed, a workaround is available, or the issue is resolved.</p>

#### Visual Studio Feedback System on 3/31/2020, 06:05 PM: 

<p>Thank you for sharing your feedback! Our teams prioritize action on product issues with broad customer impact. See details at: <a target='_blank' href="https://docs.microsoft.com/en-us/visualstudio/ide/report-a-problem?view=vs-2019#faq">https://docs.microsoft.com/en-us/visualstudio/ide/report-a-problem?view=vs-2019#faq</a>. In case you need answers to common questions or need assisted support, be sure to use <a target='_blank' href="https://visualstudio.microsoft.com/vs/support/">https://visualstudio.microsoft.com/vs/support/</a>. We’ll keep you posted on any updates to this feedback.</p>

---
### Original Solutions
(no solutions)

**Comment by mavasani** [comment]
I believe this is by design, but @CyrusNajmabadi to confirm.

**Comment by CyrusNajmabadi** [comment]
Yes, this is by design.  The cascading indicates all the items semantically linked.

Note: we're working with editor to make it possible to collape items by default so we can ensure hte primary item is at the top and expanded.
